src/utils.ts

```diff
@@ -181,10 +182,51 @@ function detectEndOfLine(text: string): "cr" | "crlf" | "lf" {
   }
 }
 
+function binaryIndexOf<T>(
```

Comment:
Is there any module about this?. I think that could be more strange if we use a module for this

Reply:
Do you mean package? Sure there is but it's only a few lines of simple code, so I just implemented it real quick. (That being said, I have implemented many variations of binary search a lot of times, so my perception may be skewed.)

Should we use a third-party package instead?

Reply:
If there is a third-party package, i suggestion you to use them.

I think you are best about that, but packages are best in feature too.